src/zarr/v3/group.py

```diff
@@ -271,8 +290,32 @@ def __repr__(self):
     async def nchildren(self) -> int:
         raise NotImplementedError
 
-    async def children(self) -> AsyncIterator[AsyncArray, AsyncGroup]:
-        raise NotImplementedError
+    async def children(self) -> AsyncGenerator[AsyncArray, AsyncGroup]:
+        """
+        Returns an AsyncGenerator over the arrays and groups contained in this group.
+        This method requires that `store_path.store` supports directory listing.
+        """
+        if not self.store_path.store.supports_listing:
+            msg = (
+                f"The store associated with this group ({type(self.store_path.store)}) "
+                "does not support listing, "
+                "specifically via the `list_dir` method. "
+                "This function requires a store that supports listing."
+            )
+
+            raise ValueError(msg)
+        subkeys = await self.store_path.store.list_dir(self.store_path.path)
```

Comment:
Following https://github.com/zarr-developers/zarr-python/discussions/1686#discussioncomment-8620826, this method should also return an async generator. Just calling that out as a change that will come here soon, what you've done is fine for now.

What I would love to see is something like:

```python
async for subkey in self.store_path.store.list_regex(self.store_path.path, pattern=pattern):
    yield await self.getitem(subkey)
```